docs/concepts/hyperparameters.md

```diff
@@ -1,5 +1,103 @@
 (hyperparameters)=
 # {octicon}`tools` Hyperparameters
 
-Explain the influence of number of epochs, batch size, for learning rate - reference to optimizer documentation
+Beyond the simplest use case,
+Finetuner gives you the flexibility to set hyper-parameters explicitly:
 
+```python
+import finetuner
+from finetuner import DocumentArray
+from finetuner.data import CSVOptions
+
+train_data = 'path/to/some/train_data.csv'
+eval_data = 'path/to/some/eval_data.csv'
+
+# Create an experiment
+finetuner.create_experiment(name='finetune-flickr-dataset')
+
+run = finetuner.fit(
+    model='efficientnet_b0',
+    train_data=train_data,
+    eval_data=eval_data, 
+    run_name='finetune-flickr-dataset-efficientnet-1',
+    description='this is a trial run on flickr8k dataset with efficientnet b0.',
+    experiment_name='finetune-flickr-dataset', # Link to the experiment created above.
+    model_options={}, # Additional options to pass to the model constructor.
+    loss='TripletMarginLoss', # Use CLIPLoss for CLIP fine-tuning.
+    miner='TripletMarginMiner',
+    miner_options={'margin': 0.2}, # Additional options for the miner constructor.
+    scheduler='linear', # Use a linear scheduler to adjust the learning rate.
+    scheduler_options={}, # Additional options for the scheduler.
+    optimizer='Adam',
+    optimizer_options={'weight_decay': 0.01}, # Additional options for the optimizer.
+    learning_rate = 1e-4,
+    epochs=5,
+    batch_size=128,
+    scheduler_step='batch',
+    freeze=False, # If applied will freeze the embedding model, only train the MLP.
+    output_dim=512, # Attach a MLP on top of embedding model.
+    device='cuda',
+    num_workers=4,
+    to_onnx=False,  # If set, please pass `is_onnx` when making inference.
+    csv_options=CSVOptions(),  # Additional options for reading data from a CSV file.
+    public=False,  # If set, anyone has the artifact id can download your fine-tuned model.
+    num_items_per_class=4,  # How many items per class to include in a batch.
+)
+```
+
+Several of these parameters are explained in detail in their own sections,
+the rest of this section will explain how best to use the ones that aren't.
+
+## Learning Rate
+The learning rate affects how much new training data affects the weights of the model you are training,
+a higher learning rate results in a greater change.
+It is important to choose an appropriate learning rate when training;
+too high and you will overfit on your training data and you will lose the knowledge gained from pre-training,
+too low and the model will not chage enough to produce significant improvement.  
+Typically, an apropriate learning rate lies in the range of `1e-4` to `1e-6`. To see a typical learning rate for the different models and tasks we support, check out our [examples](#todo: figure out what to put here).
```

Comment:
I personally feel that asking users to pick a good LR is too much. Like how could they know? I would expect to have an auto lr finder or do an ensemble of multiple runs with different lr.

 But this is outside of the scope of Documentation I guess